astropy/cosmology/flrw.py

```diff
@@ -1453,17 +1451,15 @@ class FlatFLRWMixin(FlatCosmologyMixin):
     but ``FlatLambdaCDM`` **will** be flat.
     """
 
+    Ode0 = Parameter(doc="Omega dark energy; dark energy density/critical density at z=0.",
+                     fixed=True)  # no longer a Parameter
+
     def __init__(self, *args, **kw):
         super().__init__(*args, **kw)
         # Do some twiddling after the fact to get flatness
         self._Ode0 = 1.0 - self._Om0 - self._Ogamma0 - self._Onu0
```

Comment:
My tendency would still be to go through `__set__`, i.e., `self.Ode0 = ...`

Reply:
I would, but ``Ode0`` has already been set once. So only the private attribute is writeable.

Reply:
Is that possible for people to put in `Ode0`? And then it gets overwritten? That seems a bit odd...

Reply:
Sorry, I don't think I'm following this...

If you mean passing ``Ode0`` as an argument, then all the flat classes like `FlatLambdaCDM` don't take it as an argument.
I do agree that ``FlatFLRWMixin`` should explicitly prevent this!

Reply:
I'm adding to FlatFLRWMixin

```python
def __init_subclass__(cls):
        super().__init_subclass__()
        if "Ode0" in cls._init_signature.parameters:
            raise TypeError("subclasses of `FlatFLRWMixin` cannot have `Ode0` in `__init__`")
```

Reply:
class-level protection against `Ode0`.

Reply:
I meant how is it possible for `Ode0` to already have been set here once at this place?

Reply:
E.g. ``FlatLambdaCDM.__init__()``has 
```
super().__init__(H0=H0, Om0=Om0, Ode0=0.0, Tcmb0=Tcmb0, Neff=Neff,
                         m_nu=m_nu, Ob0=Ob0, name=name, meta=meta)
```
And then ``FlatFLRWMixin`` modifies the value.

It used to be
```
super().__init__(H0=H0, Om0=Om0, Ode0=0.0, Tcmb0=Tcmb0, Neff=Neff,
                         m_nu=m_nu, Ob0=Ob0, name=name, meta=meta)
self._Ode0 = ....
```

Reply:
It seems that there should be a slightly better way... But no reason to hold this up for that!

Reply:
Agreed! I think I have an idea for a solution. Going to require a small refactor of ``Parameter`` getters, so I'll get it in before the v5.0 lock!